drivers/rtc/rtc_rv3028_eeprom.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 ANITRA system s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "rtc_rv3028.h"
+
+LOG_MODULE_DECLARE(rv3028, CONFIG_RTC_LOG_LEVEL);
+
+#define RV3028_EEBUSY_READ_POLL_MS  1
+#define RV3028_EEBUSY_WRITE_POLL_MS 10
+
+static int rv3028_eeprom_wait(const struct device *dev, int poll_ms)
+{
+	uint8_t status = 0;
+	int err;
+	int64_t timeout_time = k_uptime_get() + RV3028_EEBUSY_TIMEOUT_MS;
+
+	/* Wait while the EEPROM is busy */
+	for (;;) {
+		k_msleep(poll_ms);
+
+		err = rv3028_read_reg8(dev, RV3028_REG_STATUS, &status);
+		if (err) {
+			return err;
+		}
+
+		if (!(status & RV3028_STATUS_EEBUSY)) {
+			break;
+		}
+
+		if (k_uptime_get() > timeout_time) {
+			return -ETIME;
+		}
+	}
+
+	return 0;
+}
+
+static size_t rv3028_eeprom_get_size(const struct device *dev)
+{
+	const struct rv3028_eeprom_config *config = dev->config;
+
+	return (size_t)config->size;
+}
+
+static int rv3028_eeprom_read(const struct device *dev, off_t offset, void *data, size_t len)
+{
+	const struct rv3028_eeprom_config *config = dev->config;
+	uint8_t addr;
+	int err;
+
+	rv3028_lock_sem(dev);
+
+	err = rv3028_enter_eerd(dev);
```

Comment:
`rv3028_enter_eerd` uses busy wait. We can use sleep here

Reply:
Updated `rv3028_eeprom_wait_busy` to use `k_msleep` if the kernel is initialized:
https://github.com/zephyrproject-rtos/zephyr/blob/186acfc1ffbc3d17182d15d3091d8e5042e785d2/drivers/rtc/rtc_rv3028.c#L206-L210

Reply:
Updated `rv3028_eeprom_wait_busy` to use `k_msleep`
https://github.com/zephyrproject-rtos/zephyr/blob/e8bee4676e172ce1ac97965da7cc68d1e559bd34/drivers/rtc/rtc_rv3028.c#L206